src/Interpreters/DatabaseCatalog.h

```diff
@@ -245,6 +245,7 @@ class DatabaseCatalog : boost::noncopyable, WithMutableContext
 
     void shutdownImpl();
 
+    static void checkTableCanBeRemovedOrRenamedImpl(const TableNamesSet & dependent, const QualifiedTableName & removing_table, bool is_drop_database);
```

Comment:
How about calling this `checkTableDependencies` ?  as we call it even outside of `checkTableCanBeRemovedOrRenamed`

Reply:
But it's not just "checking dependencies" (whatever it means), it checks if table can be removed or renamed. So I think that `checkTableCanBeRemovedOrRenamedImpl` is more clear